os/drivers/wireless/virtual/vdev.c

```diff
@@ -213,13 +216,109 @@ int _vwifi_create_msgqueue(int *fd)
 	return 0;
 }
 
+/*
+ * Internal Function
+ */
+static struct netdev* vdev_register_dev(int sizeof_priv)
+{
+	struct nic_io_ops nops = {vdev_linkoutput, vdev_set_multicast_list};
+	struct netdev_config nconfig;
+	nconfig.ops = &nops;
+	nconfig.flag = NM_FLAG_ETHARP | NM_FLAG_ETHERNET | NM_FLAG_BROADCAST | NM_FLAG_IGMP;
+	nconfig.mtu = CONFIG_NET_ETH_MTU; // is it right that vendor decides MTU size??
+	nconfig.hwaddr_len = IFHWADDRLEN;
+	nconfig.is_default = 1;
+
+	nconfig.type = NM_WIFI;
+	nconfig.t_ops.wl = &g_trwifi_drv_ops;
+	nconfig.priv = NULL;
+
+	return netdev_register(&nconfig);
+}
+
+static void vwifi_send_packet(uint8_t *buf, uint32_t len)
+{
+	if (!g_vwifi_dev) {
+		VWIFI_ERROR(0);
+		return;
+	}
+	netdev_input(g_vwifi_dev, buf, len);
+}
+
+/*
+ * Handler
+ */
+static void vdev_run(int argc, char *argv[])
+{
+	g_vwifi_dev = vdev_register_dev(0);
+	if (!g_vwifi_dev) {
+		VWIFI_ERROR(0);
+		return;
+	}
+	netdev_set_hwaddr(g_vwifi_dev, g_hwaddr, IFHWADDRLEN);
+
+	int fd;
+	int res = _vwifi_create_msgqueue(&fd);
+	if (res < 0) {
+		VWIFI_ERROR(0);
+		return;
+	}
+
+	fd_set rfds, tfds;
+	FD_ZERO(&rfds);
+	FD_SET(fd, &rfds);
+
+	while (1) {
+		tfds = rfds;
+		res = select(fd + 1, &tfds, NULL, NULL, NULL);
+		if (res <= 0) {
+			VWIFI_ERROR(res);
+			if (errno == EINTR) {
+				continue;
+			}
+			break;
+		}
+		struct vwifi_msg msg;
+		res = _recv_message(fd, (char *)&msg, sizeof(struct vwifi_msg));
+		if (res < 0) {
+			VWIFI_ERROR(res);
+			break;
+		}
+
+		res = vwifi_handle_message(msg.req);
+		if (res < 0) {
+			VWIFI_ERROR(res);
+			break;
+		}
+		_send_signal(&msg);
+	}
+	return;
```

Comment:
I think this return is not needed.

Reply:
I think this is trivial. so I won't remove it